pkg/extensions/extensions.go

```diff
@@ -34,6 +34,39 @@ func downloadTrivyDB(dbDir string, log log.Logger, updateInterval time.Duration)
 	}
 }
 
+// Scrub ...
+func scrub(log log.Logger, scrubInterval time.Duration, storeController storage.StoreController) error {
+	for {
+		log.Info().Msg("executing scrub to check manifest/blob integrity")
+
+		result, err := storeController.CheckAllBlobsIntegrity()
```

Comment:
The worry is whether this holds a lock for a long time. A simple test is to have many repos and then try to push an image when a scrub is ongoing, will that push hang? You need a generator pattern: https://wiki.python.org/moin/Generators

Reply:
Also don't you want to put his in scrub/scrub.go instead of here?